framework/client.py

```diff
@@ -154,5 +155,7 @@ def set_user_agent(self, ua):
         self.options.append('-H \'User-Agent: %s\'' % ua)
 
     def wait_for_finish(self):
-        self.proc.join()
+        # until we explicitly get `self.exit_event` flag set
+        while self.is_busy(verbose=False):
```

Comment:
Can't we use `self.exit_event.wait(some timeout)` here?

Reply:
Internally [it](https://github.com/python/cpython/blob/89f52293281b6efc4ef666ef25e677683821f4b9/Lib/threading.py#L288) uses locks and does some extra job, but we're not setting that event flag in different places, hence we don't really need it.